package.json

```diff
@@ -113,16 +113,17 @@
     "style-loader": "^0.13.2",
     "styled-components": "^4.0.3",
     "typeface-lato": "^0.0.75",
-    "webpack": "^4.30.0",
-    "webpack-bundle-analyzer": "^3.3.2",
-    "webpack-dev-middleware": "^3.1.3",
-    "webpack-hot-middleware": "^2.24.3",
+    "webpack": "^5.74.0",
```

Comment:
[using this code line simply in order to start a thread]

I tested the extension architecture by building nextstrain.org with this version of auspice (see below). I remember the extension-architecture was difficulat as it involved lots of webpack configuration which (at the time) didn't have the best docs. I'd be happy to help debug this stuff with you, although you're probably the webpack expert after this PR :)

P.S. I haven't tested auspice.us, but that should be done.


```
## working directory: nextstrain.org
$ npm install nextstrain/auspice#9c1df8bcc7f42733077343218fb0e8e5003c704c
$ cd auspice-client/
$ ../node_modules/.bin/auspice build --verbose --extend ./customisations/config.json
[verbose]       Generating webpack config. Extensions? true. devMode: false
[verbose]       Webpack writing output to: /Users/enigma/projects/nextstrain/nextstrain.org/auspice-client/dist
Running webpack compiler
[verbose]       Generating Babel Config
[BABEL] Note: The code generator has deoptimised the styling of /Users/enigma/projects/nextstrain/nextstrain.org/node_modules/react-icons/fa/index.esm.js as it exceeds the max of 500KB.
assets by status 2.82 MiB [cached] 44 assets
Entrypoint main = auspice.core-vendors.bundle.d772f646f5d6159b105d.js auspice.main.bundle.efb7ad8e25040a23a1df.js 3 auxiliary assets
orphan modules 1.37 MiB [orphan] 518 modules
runtime modules 8.81 KiB 14 modules
javascript modules 7.12 MiB
  modules by path ./ 2.26 MiB
    modules by path ./node_modules/core-js/ 683 KiB 701 modules
    modules by path ./src/ 1.57 MiB
      cacheable modules 1.57 MiB 89 modules
      modules by path ./src/locales/ 382 bytes 2 modules
    ./node_modules/regenerator-runtime/runtime.js 24.3 KiB [built] [code generated]
  modules by path ../ 4.86 MiB 359 modules
  fs (ignored) 15 bytes [built] [code generated]
json modules 57.8 KiB 37 modules

ERROR in ../../auspice-client/customisations/languageSelector.js 33:0-44
Module not found: Error: Can't resolve 'iso-639-1/build/index' in '/Users/enigma/projects/nextstrain/nextstrain.org/auspice-client/customisations'
Did you mean 'index.js'?
BREAKING CHANGE: The request 'iso-639-1/build/index' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
 @ ../../auspice-client/customisations/ sync ^\.\/.*$ ./languageSelector.js ./languageSelector
 @ ./src/util/extensions.js 13:24-71
 @ ./src/util/googleAnalytics.js 10:0-58 19:16-28 44:31-43
 @ ./src/index.js 18:0-77 45:0-35
 @ ./src/indexAsync.js

ERROR in ../../auspice-client/customisations/navbar.js 38:0-50
Module not found: Error: Can't resolve './languageSelector' in '/Users/enigma/projects/nextstrain/nextstrain.org/auspice-client/customisations'
Did you mean 'languageSelector.js'?
BREAKING CHANGE: The request './languageSelector' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
 @ ../../auspice-client/customisations/ sync ^\.\/.*$ ./navbar.js ./navbar
 @ ./src/util/extensions.js 13:24-71
 @ ./src/util/googleAnalytics.js 10:0-58 19:16-28 44:31-43
 @ ./src/index.js 18:0-77 45:0-35
 @ ./src/indexAsync.js

ERROR in ../../auspice-client/customisations/splash.js 25:0-69
Module not found: Error: Can't resolve 'auspice/src/components/markdownDisplay' in '/Users/enigma/projects/nextstrain/nextstrain.org/auspice-client/customisations'
Did you mean 'index.js'?
BREAKING CHANGE: The request 'auspice/src/components/markdownDisplay' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
 @ ../../auspice-client/customisations/ sync ^\.\/.*$ ./splash.js ./splash
 @ ./src/util/extensions.js 13:24-71
 @ ./src/util/googleAnalytics.js 10:0-58 19:16-28 44:31-43
 @ ./src/index.js 18:0-77 45:0-35
 @ ./src/indexAsync.js

ERROR in ../../auspice-client/customisations/splash.js 26:0-30
Module not found: Error: Can't resolve './navbar' in '/Users/enigma/projects/nextstrain/nextstrain.org/auspice-client/customisations'
Did you mean 'navbar.js'?
BREAKING CHANGE: The request './navbar' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
 @ ../../auspice-client/customisations/ sync ^\.\/.*$ ./splash.js ./splash
 @ ./src/util/extensions.js 13:24-71
 @ ./src/util/googleAnalytics.js 10:0-58 19:16-28 44:31-43
 @ ./src/index.js 18:0-77 45:0-35
 @ ./src/indexAsync.js

4 errors have detailed information that is not shown.
Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.

webpack 5.74.0 compiled with 4 errors in 31508 ms
[error] Webpack built with errors. Exiting.
```

Reply:
This works on my machine now, but I've created some PRs to test this in a more robust way: https://github.com/nextstrain/auspice/pull/1565, https://github.com/nextstrain/nextstrain.org/pull/599

Reply:
This also works on my machine - both the source maps and the hot reloading are working as expected 😄

Reply:
I can still reproduce my errors above via the following (now using the latest commit)

```
npm install nextstrain/auspice#6b9fddf0e187ee5bf9f2bd9537eb0eee164156d2
cd auspice-client
../node_modules/.bin/auspice build --verbose --extend ./customisations/config.json
```

Reply:
You're right, the same errors show up in the [preview PR build](https://github.com/nextstrain/nextstrain.org/actions/runs/3527897039/jobs/5917442431).

My first thought was that this is related to https://github.com/nextstrain/nextstrain.org/issues/616, but the presence of successful preview PR builds that came after the ESM conversion ([example](https://github.com/nextstrain/nextstrain.org/actions/runs/3432471538/jobs/5721781208)) rules that out.

A search online shows that it is indeed related to the webpack 5 upgrade: https://github.com/webpack/webpack/issues/11467

I'm going to investigate this some more tomorrow.

Reply:
This is a simple fix on the auspice-client side: https://github.com/nextstrain/nextstrain.org/pull/634

I've noted in this PR's description that the above PR should be merged first.